test/adapter/json/polymorphic_test.rb

```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class Json
+        class PolymorphicTest < ActiveSupport::TestCase
+          setup do
+            @employee = Employee.new(id: 42, name: 'Zoop Zoopler', email: 'zoop@example.com')
+            @picture = @employee.pictures.new(id: 1, title: 'headshot-1.jpg')
+            @picture.imageable = @employee
+
+            @serialization = serializable(@picture, adapter: :json)
+            ActionController::Base.cache_store.clear
+          end
+
+          def test_polymorphic_serializer
+            assert_equal({ id: 42, name: 'Zoop Zoopler' }, @serialization.as_json[:picture][:imageable])
+          end
```

Comment:
Thanks for being responsive... I know thr async of this can be a bit annoying when there's a lot of back and forth... i really appreciate your brjngjng this up and working on is documenting the behavior and even making us thing about it